### ISSUE TITLE
Lambda Layers create incorrect output_shape when used on array of Tensors

### ISSUE THREAD

**Issue by kuza55** [fchollet/keras#3599]
This code crashes with this output:
Exception: Only layers of same output shape can be merged using sum mode. Layer shapes: [((None, 10), 10), (None, 10)]

```
import keras
from keras.layers import Input, Dense, merge
from keras.models import Model
from keras.layers.core import Lambda

def dense_fn(data_arr, output_dims):
    out_arr = []
    for layer in data_arr:
        out_dense = Dense(output_dims)(layer)
        out_arr.append(out_dense)

    return merge(out_arr, mode='sum')

numeric_columns = 10
input = Input(shape=(numeric_columns,), name='numeric_input')

concat_layers = []
hidden = input
concat_layers.append(hidden)
hidden = Dense(10)(hidden)
concat_layers.append(hidden)
hidden = Lambda(dense_fn, output_shape=(10,), arguments={'output_dims':10})(concat_layers)
concat_layers.append(hidden)
hidden = merge([hidden, input])

model = Model(input=input, output=hidden)
```

I spent some time tracking it down and it seems to be due to the Lambda layer not dealing with lists of inputs. In TensorFlow, simply removing the output_shape parameter fixes this issue since TensorFlow figures out the correct shape.

I'm not really sure what the right fix for this is; if the architecture doesn't support it, then maybe just adding an exception saying that output_shape and arrays of inputs are incompatible?

**Comment by farizrahman4u** [comment]
If you have `Dense` layers inside a `Lambda` layer's function, remember that the weights of the `Dense` layer will not be trained. They will be frozen with their initial random weights. What are you trying to do here?

**Comment by kuza55** [comment]
Oh, thanks for the pointer. That would explain. Why it didn't train well. The code above is just a repro for the bug though.

In my actual network I'm trying to play with the ideas from this paper: http://arxiv.org/abs/1608.06993 but without explicit concatenations.

**Comment by farizrahman4u** [comment]
When you have a list of tensors, you use the Merge layer (or merge function). You can use the mode argument to specify your function.

**Comment by farizrahman4u** [comment]
Not a bug, Lambda layer accepts only a single tensor.

**Comment by niowniow** [comment]
@farizrahman4u There might actually be a bug!? When I run the "The Merge layer" example from the Keras documentation (Guide to the Sequential model) I get 

`TypeError: <lambda>() missing 1 required positional argument: 'y'`

The code is the same from the example with this merge layer:

`merged = Merge([left_branch, right_branch], mode=lambda x, y: x - y)`

I'm using up-to-date Theano & Keras. I'm not sure though, if this is related to the OP's problem

**Comment by farizrahman4u** [comment]
The documentation is wrong.
```python
merged = Merge([left_branch, right_branch], mode=lambda x: x[0] - x[1])
```

**Comment by farizrahman4u** [comment]
I have made a PR to fix the docs : #3630
